src/base/http/connection.cpp

```diff
@@ -45,8 +45,18 @@ Connection::Connection(QTcpSocket *socket, IRequestHandler *requestHandler, QObj
     , m_requestHandler(requestHandler)
 {
     m_socket->setParent(this);
+
+    // reset timer when there are activity
     m_idleTimer.start();
-    connect(m_socket, &QTcpSocket::readyRead, this, &Connection::read);
+    connect(m_socket, &QIODevice::readyRead, this, [this]()
+    {
+        m_idleTimer.start();
```

Comment:
Why not `restart()` as it was used before?

Reply:
https://doc.qt.io/qt-5/qelapsedtimer.html#restart

They are mostly equivalent and I suppose `start()` is a bit faster since it don't return elapsed time.